### PR TITLE
Add BuildFileType metric telemetry

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#193620](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/193620)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5688)

### CODE REVIEW

File: src/docfx/lib/log/Telemetry.cs [hunk 1]
```diff
@@ -65,26 +75,25 @@ public static void TrackCacheMissCount(TelemetryName name)
             s_cacheCountMetric.TrackValue(1, name.ToString(), "miss", s_os, s_version, s_repo, s_branch, s_correlationId);
         }
 
-        public static void TrackBuildItemCount(ContentType contentType)
+        public static void TrackBuildFileTypeCount(Document file)
         {
-            s_buildItemCountMetric.TrackValue(1, $"{TelemetryName.BuildItems}-{contentType}", s_os, s_version, s_repo, s_branch, s_correlationId);
+            var fileExtension = (Path.GetExtension(file.FilePath.Path) ?? "<null>").ToLowerInvariant();
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs]:
```suggestion
            var fileExtension = file.FilePath.Format.ToString();
```

File: src/docfx/lib/log/Telemetry.cs [hunk 2]
```diff
@@ -24,7 +24,8 @@ internal static class Telemetry
         private static readonly Metric s_operationTimeMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Time", "Name", "OS", "Version", "Repo", "Branch", "CorrelationId"), s_metricConfiguration);
         private static readonly Metric s_errorCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"BuildLog", "Code", "Level", "Type", "OS", "Version", "Repo", "Branch", "CorrelationId"), s_metricConfiguration);
         private static readonly Metric s_cacheCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Cache", "Name", "State", "OS", "Version", "Repo", "Branch", "CorrelationId"), s_metricConfiguration);
-        private static readonly Metric s_buildItemCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Count", "Name", "OS", "Version", "Repo", "Branch", "CorrelationId"), s_metricConfiguration);
+        private static readonly Metric s_buildCommitCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"Count", "Name", "OS", "Version", "Repo", "Branch", "CorrelationId"), s_metricConfiguration);
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs]: If `os`, `version`, `repo`, `branch` become global properties, can they be removed from dimensions?

Comment (928PJY) [reply src/docfx/lib/log/Telemetry.cs]: There is a tricky thing in the telemetry library:
The global property will not be applied to metric if we use the metric container to do the pre-aggregate.

File: src/docfx/lib/log/Telemetry.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/lib/log/Telemetry.cs]: the format just support `json`, `md` and `yml`, what about resource? like `.jpg`

File: src/docfx/lib/log/Telemetry.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/lib/log/Telemetry.cs]: hmmm, then seems the original code is more clear?

File: src/docfx/lib/log/Telemetry.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/lib/log/Telemetry.cs]: yep you are right.
```suggestion
            var fileExtension = Path.GetExtension(file.FilePath.Path)?.ToLowerInvariant() ?? "<null>";
```

File: src/docfx/lib/log/Telemetry.cs (hunk 2 above)
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs]:
```suggestion
        private static readonly Metric s_buildCommitCountMetric = s_telemetryClient.GetMetric(new MetricIdentifier(null, $"BuildCommitCount", "Name", "OS", "Version", "Repo", "Branch", "CorrelationId"), s_metricConfiguration);
```

Comment (928PJY) [reply src/docfx/lib/log/Telemetry.cs]: The global property still works for `TrackEvent` `TrackMetric` and `TrackException`

File: src/docfx/lib/log/Telemetry.cs [hunk 3]
```diff
@@ -34,10 +35,19 @@ internal static class Telemetry
 
         private static string s_correlationId = EnvironmentVariable.CorrelationId ?? Guid.NewGuid().ToString("N");
 
+        static Telemetry()
+        {
+            s_telemetryClient.Context.GlobalProperties["OS"] = RuntimeInformation.OSDescription ?? "<null>";
+            s_telemetryClient.Context.GlobalProperties["Version"] = typeof(Telemetry).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "<null>";
+            s_telemetryClient.Context.GlobalProperties["CorrelationId"] = EnvironmentVariable.CorrelationId ?? Guid.NewGuid().ToString("N");
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs]: Ok, if this to be put into `GlobalProperties`, ensure it is using the same value:
```suggestion
            s_telemetryClient.Context.GlobalProperties["CorrelationId"] = s_correlationId;
```

Consider doing this for other global properties

**Pull request merged**